### PR TITLE
feat: pass tokens into requests and utilize token rotation

### PR DESCRIPTION
**Pull request by jiejasonliu**
### Changes
- landing page `sign in` button just does a dummy login
- tests are consolidated to use `renderWithTestRoots` (or `withTestRoots` for hooks) if a router/recoil is needed
- header now shows a `logout` button if the user is logged in (auth jwt exists in local storage)
- fetch wrapper will attempt to refresh the token on the first HTTP 401
  - also slight refactor, fetch wrapper now uses pseudo "middlewares" 
- add `AuthorizedGuardLayout` to guard routes that need user authentication (so all our feature pages)

---

#### (Header) logout button
image

### CODE REVIEW

**Review by derricklee43 — APPROVED** [review]
lgtm

File: src/hooks/api/use-user-client.ts [hunk 1]
```diff
@@ -18,26 +18,33 @@ export function useUserClient() {
...
       // TODO: when token refreshing is implemented, this should probably be an object
       // that contains both the access token as well as the refresh token
-      const { accessToken } = await fetchWrapper.post('/Authenticate', payload, numRetries);
+      const jwtData = await fetchWrapper.post('/Authenticate', payload, numRetries);
       const authJwt = {
-        accessToken: accessToken,
+        accessToken: jwtData,
         refreshToken: undefined,
       };
+      debugger;
```
[the remark this answers is not in the document]
Comment (jiejasonliu) [reply src/hooks/api/use-user-client.ts]: nope, gone now 😄

**Comment by jiejasonliu** [comment]
> lgtm

i added 3 commits to utilize token rotation (so the refresh token too)
i've also updated the description, can you take another look?

**Comment by derricklee43** [comment]
yep, still looks good!

**Pull request merged**